### ISSUE TITLE
Week 11 Update

### ISSUE THREAD

**Issue by SidhantPuntambekar** [SidhantPuntambekar/BMI702_Final_Project#10]
Latest Commit: f8b869780204e1e78822ace17e7ac2a4ecac363a

What we did this week:

- Verified code implementation for fine-tuning AbLang2 for approval prediction and developability metrics.
- Update fine-tuned AbLang2 with GDPa1 and LoRA for approval prediction and developability metric regression with ridge classifier head. LoRA performance seems to be worse compared to frozen AbLang2 with ridge head.
- Worked in fine-tuned AbLang2 with LoRA code for generative pipeline (https://github.com/SidhantPuntambekar/BMI702_Final_Project/pull/9). Still need to do some checking this deeply before merging.
- Instead of only using Thera-SabDab, tried downloading the full SAbDab to fetch all the CDR3 regions and their antigens.
- Tried implementing cross attention binding scorer on antigens and antibody CDR3 regions using contrastive pretraining
- Then tried fine tuning on our therapeutic antibody and use guided ODE on 480d CDR3 embeddings from the current 10d PLS position (instead of working on 10d PLS space and 1280 antigen ESM2 embedding space by concatinating them)

What's next:

- Set up the the evaluation pipeline for generative model using developability prediction fine-tuned from AbLang2 (e.g, calculating delta delta G)
- Pretrain flow matched model if time
- Start working on slides and final report
- fix error on calling Sabdab csv file, if doesn't work, simply download through their website
- rerun the full generative pipeline
